### PR TITLE
CSHARP-4098: Include more detail when throwing PlatformNotSupportedException errors

### PR DESCRIPTION
**Pull request by BorisDog**

### CODE REVIEW

File: Docs/reference/content/reference/driver/crud/compression.md [hunk 1]
```diff
@@ -12,9 +12,9 @@ title = "Compression"
 
 The C# driver supports compression of messages to and from MongoDB servers. The driver implements the three algorithms that are supported by MongoDB servers:
 
-* [Snappy](https://google.github.io/snappy/): Snappy compression can be used when connecting to MongoDB servers starting with the 3.4 release.
-* [Zlib](https://zlib.net/): Zlib compression can be used when connecting to MongoDB servers starting with the 3.6 release.
-* [Zstandard](https://facebook.github.io/zstd/): Zstandard compression can be used when connecting to MongoDB servers starting with the 4.2 release.
+* [Snappy](https://google.github.io/snappy/): Snappy compression can be used when connecting to MongoDB servers starting with the 3.4 release. Supported platforms: Windows: x32 and x64, Linux/macOS: x64.
```
Comment (DmitryLukyanov) [inline Docs/reference/content/reference/driver/crud/compression.md:15]: It probably makes sense to mention absence for arm support?

File: src/MongoDB.Driver.Core/Core/Compression/Snappy/Snappy32NativeMethods.cs [hunk 2]
```diff
@@ -107,12 +107,13 @@ private class Delegates32
...
         {
             public override bool IsX32ModeSupported => OperatingSystemHelper.CurrentOperatingSystem == OperatingSystemPlatform.Windows;
+            public override string LibraryName => "Snappy";
 
             public override string GetLibraryName(OperatingSystemPlatform currentPlatform) =>
                 currentPlatform switch
                 {
                     OperatingSystemPlatform.Windows => "snappy32.dll", // supported only on windows
-                    _ => throw new InvalidOperationException($"Snappy is not supported on the current platform: {currentPlatform} in 32-bit mode."),
+                    _ => throw new InvalidOperationException($"{LibraryName} is not supported on the current platform: {currentPlatform} in 32-bit mode."),
```
Comment (DmitryLukyanov) [inline src/MongoDB.Driver.Core/Core/Compression/Snappy/Snappy32NativeMethods.cs:116]: it looks like this (and the rest) check can be moved to `RelativeLibraryLocatorBase` (with using IsX32ModeSupported)?

Comment (DmitryLukyanov) [reply src/MongoDB.Driver.Core/Core/Compression/Snappy/Snappy32NativeMethods.cs:116]: Also, it looks like having both `LibraryName` and `GetLibraryName` at the same time confuse. I think adding LibraryName property can be avoided

Comment (BorisDog) [reply src/MongoDB.Driver.Core/Core/Compression/Snappy/Snappy32NativeMethods.cs:116]: Renamed `GetLibraryName` to `GetLibraryFileName`

File: Docs/reference/content/reference/driver/crud/compression.md (hunk 1 above)
Comment (BorisDog) [reply Docs/reference/content/reference/driver/crud/compression.md]: Discussed offline, decided to mention only supported platforms.

**Review by DmitryLukyanov — APPROVED** [review]
LGTM

**Pull request merged**